wordpress-mobile/WordPress-iOS
# Fix: Undo action restores Gutenberg template content as Classic editor

**Fixes** https://github.com/wordpress-mobile/gutenberg-mobile/issues/3342
**Related PR:** https://github.com/wordpress-mobile/WordPressKit-iOS/pull/385

## Description
This addresses https://github.com/wordpress-mobile/gutenberg-mobile/issues/3342 by pulling in the changes in the related PR. ~It also addresses an issue I noticed while testing. After restoring a deleted post, the post would be converted to a draft but it remained in the published list. This adds a refresh similar to the way the delete works to make sure the list is refreshed.~

Refreshing the list was separated to https://github.com/wordpress-mobile/WordPress-iOS/pull/16358

## To test:

1. Create a site that generates posts such as Cassel under the Blog category. 
    * This is an important step because for some reason that I wasn't able to track down logging in on a different device restores the content as a Gutenberg Post without then added parameter.
2. Navigate to a Post and ensure it's a Gutenberg Post.
3. Delete it. 
4. Restore it. 
5. Validate that the restored post ~(now in drafts)~ uses the Gutenberg Editor. 

## Regression Notes
1. Potential unintended areas of impact
This should be the only affected flow.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N/A

3. What automated tests I added (or what prevented me from doing so)
Refactoring to establish tests would dramatically increase the size of this effort.

## PR submission checklist:
